integration-test/src/test/resources/test_python_api.py

```diff
@@ -0,0 +1,223 @@
+#
+# Licensed to Cloudera, Inc. under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  Cloudera, Inc. licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import base64
+import json
+import time
+from urlparse import urlparse
+import requests
+import cloudpickle
+import pytest
+import httplib
+from flaky import flaky
+
+global session_id, job_id
+session_id = None
+job_id = None
+
+livy_end_point = os.environ.get("LIVY_END_POINT")
+add_file_url = os.environ.get("ADD_FILE_URL")
+add_pyfile_url = os.environ.get("ADD_PYFILE_URL")
+upload_file_url = os.environ.get("UPLOAD_FILE_URL")
+upload_pyfile_url = os.environ.get("UPLOAD_PYFILE_URL")
+
+
+@pytest.fixture(scope="module", autouse=True)
+def after_all(request):
+    request.addfinalizer(stop_session)
+
+
+def process_job(job, expected_result, is_error_job=False):
+    global job_id
+
+    pickled_job = cloudpickle.dumps(job)
+    base64_pickled_job = base64.b64encode(pickled_job).decode('utf-8')
+    base64_pickled_job_json = json.dumps({'job': base64_pickled_job})
+    request_url = livy_end_point + "/sessions/" + str(session_id) + "/submit-job"
+    header = {'Content-Type': 'application/json', 'X-Requested-By': 'livy'}
+    response = requests.request('POST', request_url, headers=header, data=base64_pickled_job_json)
+
+    assert response.status_code == httplib.CREATED
+    job_id = response.json()['id']
+
+    poll_time = 0.1
+    max_poll_time = 10
+    poll_response = None
+    while (poll_response is None or poll_response.json()['state'] == 'STARTED') and poll_time < \
+            max_poll_time:
+        time.sleep(poll_time)
+        poll_request_uri = livy_end_point + "/sessions/" + str(session_id) + \
+                           "/jobs/" + str(job_id)
+        poll_header = {'X-Requested-By': 'livy'}
+        poll_response = requests.request('GET', poll_request_uri, headers=poll_header)
+        poll_time *= 2
+
+    assert poll_response.json()['id'] == job_id
+    assert poll_response.status_code == httplib.OK
+    if not is_error_job:
+        result = poll_response.json()['result']
```

Comment:
This code made it a little hard to debug things when the call actually failed unexpectedly.

Can you add an assert that in this case the error is none? That way if there's an error it will show up in the log file.